pkg/cache/azure_cache.go

```diff
@@ -120,8 +122,14 @@ func (t *TimedCache) getInternal(key string) (*AzureCacheEntry, error) {
 	return newEntry, nil
 }
 
-// Get returns the requested item by key.
+// Get returns the requested item by key with deep copy.
 func (t *TimedCache) Get(key string, crt AzureCacheReadType) (interface{}, error) {
+	data, err := t.get(key, crt)
+	copied := deepcopy.Copy(data)
```

Comment:
any specifical reason why you want to do this? @lzhecheng 
we found there is OOM in large scale test, every time it ends up copying the VMSS cache whenever a single VM is gotten from it

Reply:
It's for thread safety.